python/ray/data/_internal/execution/operators/base_physical_operator.py

```diff
@@ -79,6 +80,13 @@ def num_outputs_total(self) -> Optional[int]:
             else self.input_dependencies[0].num_outputs_total()
         )
 
+    def num_output_rows_total(self) -> Optional[int]:
+        return (
+            self._output_rows
+            if self._output_rows
+            else self.input_dependencies[0].num_output_rows_total()
```

Comment:
Is `self.input_dependencies[0].num_output_rows_total()` something that is static? Should we cache this value with some call like `self._output_rows = self.input_dependencies[0].num_output_rows_total()`?

If this total is a live total that is updated as execution continues makes sense to leave as is.

Reply:
Right! Here the `self._output_rows` is not static, but it's our primary option, as it will be update here
![image](https://github.com/user-attachments/assets/6993e35a-6b0f-4eb5-b24d-8a539bc1396d)